src-java/floodlight-service/floodlight-modules/src/main/java/org/openkilda/floodlight/kafka/ZooKeeperHandler.java

```diff
@@ -0,0 +1,101 @@
+/* Copyright 2020 Telstra Open Source
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License");
+ *   you may not use this file except in compliance with the License.
+ *   You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   Unless required by applicable law or agreed to in writing, software
+ *   distributed under the License is distributed on an "AS IS" BASIS,
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ */
+
+package org.openkilda.floodlight.kafka;
+
+import org.openkilda.bluegreen.LifeCycleObserver;
+import org.openkilda.bluegreen.LifecycleEvent;
+import org.openkilda.bluegreen.Signal;
+import org.openkilda.bluegreen.ZkStateTracker;
+import org.openkilda.bluegreen.ZkWatchDog;
+import org.openkilda.bluegreen.ZkWriter;
+
+import lombok.Getter;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.zookeeper.KeeperException;
+
+import java.io.IOException;
+import java.util.UUID;
+
+@Slf4j
+public class ZooKeeperHandler implements Runnable, LifeCycleObserver {
+
+    public static final String ZK_COMPONENT_NAME = "floodlight";
+
+    @Getter
+    private volatile LifecycleEvent event;
+
+    private ZkStateTracker zkStateTracker;
+
+    private long messageId = 0;
+    private boolean awaitingShutdown = false;
+
+    public ZooKeeperHandler(String id, String serviceName, String connectionString) {
+        try {
+            ZkWatchDog watchDog = ZkWatchDog.builder().id(id).serviceName(serviceName)
+                    .connectionString(connectionString).build();
+            watchDog.subscribe(this);
+
+            ZkWriter zkWriter = ZkWriter.builder().id(id).serviceName(serviceName)
+                    .connectionString(connectionString).build();
+            zkStateTracker = new ZkStateTracker(zkWriter);
```

Comment:
We have race condition here. 
Timeline:
1. we initialize watchDog and call subscribe
2. watch dog run new thread, read ZK and call `handle()`
3. zkStateTracker still and handle() fails because of NPE
4. We initialize zkStateTracker in main thread

Reply:
fixed